execution/function/operator.go

```diff
@@ -38,30 +38,53 @@ func NewFunctionOperator(funcExpr *parser.Call, nextOps []model.VectorOperator,
 
 	switch funcExpr.Func.Name {
 	case "scalar":
-		return &scalarFunctionOperator{
+		s := &scalarFunctionOperator{
 			next: nextOps[0],
 			pool: model.NewVectorPoolWithSize(stepsBatch, 1),
-		}, nil
+		}
+		s.OperatorTelemetry = &model.NoopTimingInformation{}
+		if opts.EnableAnalysis {
+			s.OperatorTelemetry = &model.TimingInformation{}
+		}
+		return s, nil
+
 	case "label_join", "label_replace":
-		return &relabelFunctionOperator{
+		r := &relabelFunctionOperator{
 			next:     nextOps[0],
 			funcExpr: funcExpr,
-		}, nil
+		}
+		r.OperatorTelemetry = &model.NoopTimingInformation{}
```

Comment:
You can implement this logic once and then pass it to each operator. This way you would avoid copy/paste.